dialogue-annotations/src/main/java/com/palantir/dialogue/annotations/FormUrlEncodedSerializerFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations;
+
+import com.palantir.dialogue.RequestBody;
+import java.util.Map;
+
+/**
+ * Extend this class in order to serialize {@link T} to {@code application/x-www-form-urlencoded} request body.
+ *
+ * @param <T> type of value to serialize
+ */
+public abstract class FormUrlEncodedSerializerFactory<T> extends StdSerializer<T> {
```

Comment:
I’d prefer composition over inheritance for this sort of thing, but the abstract class is probably most obvious for consumers.

Reply:
I initially just pushed the FormUrlEncodedRequestBody, but then realized that everyone will need this snippet so might as well.